Lib/test/test__colorize.py

```diff
@@ -0,0 +1,134 @@
+                f'    a {boldr}${reset} b\n',
+                f'      {boldr}^{reset}\n',
+                f'{boldm}SyntaxError{reset}: {magenta}invalid syntax{reset}\n',
+            ]
+        )
+        self.assertIn(expected, actual)
+
+    def test_colorized_traceback_is_the_default(self):
+        def foo():
+            1/0
+
+        from _testcapi import exception_print
+
+        try:
+            foo()
+            self.fail("No exception thrown.")
+        except Exception as e:
+            with captured_output("stderr") as tbstderr:
+                with unittest.mock.patch('_colorize.can_colorize', return_value=True):
+                    exception_print(e)
+            actual = tbstderr.getvalue().splitlines()
+
+        red = _colorize.ANSIColors.RED
+        boldr = _colorize.ANSIColors.BOLD_RED
+        magenta = _colorize.ANSIColors.MAGENTA
+        boldm = _colorize.ANSIColors.BOLD_MAGENTA
+        reset = _colorize.ANSIColors.RESET
+        lno_foo = foo.__code__.co_firstlineno
+        expected = [
+            'Traceback (most recent call last):',
+            f'  File {magenta}"{__file__}"{reset}, '
+            f'line {magenta}{lno_foo+6}{reset}, in {magenta}test_colorized_traceback_is_the_default{reset}',
+            f'    {red}foo{reset+boldr}(){reset}',
+            f'    {red}~~~{reset+boldr}^^{reset}',
+            f'  File {magenta}"{__file__}"{reset}, '
+            f'line {magenta}{lno_foo+1}{reset}, in {magenta}foo{reset}',
+            f'    {red}1{reset+boldr}/{reset+red}0{reset}',
+            f'    {red}~{reset+boldr}^{reset+red}~{reset}',
+            f'{boldm}ZeroDivisionError{reset}: {magenta}division by zero{reset}',
+        ]
+        self.assertEqual(actual, expected)
+
+    def test_colorized_detection_checks_for_environment_variables(self):
+        if sys.platform == "win32":
+            virtual_patching = unittest.mock.patch(
+                "nt._supports_virtual_terminal", return_value=True
+            )
+        else:
+            virtual_patching = contextlib.nullcontext()
+
+        env_vars_expected = [
+            ({'TERM': 'dumb'}, False),
+            ({'PYTHON_COLORS': '1'}, True),
+            ({'PYTHON_COLORS': '0'}, False),
+            ({'NO_COLOR': '1'}, False),
+            ({'NO_COLOR': '1', "PYTHON_COLORS": '1'}, True),
+            ({'FORCE_COLOR': '1'}, True),
+            ({'FORCE_COLOR': '1', 'NO_COLOR': '1'}, False),
+            ({'FORCE_COLOR': '1', "PYTHON_COLORS": '0'}, False),
+        ]
+
+        with virtual_patching:
+            with unittest.mock.patch("os.isatty") as isatty_mock:
+                isatty_mock.return_value = True
+
+                for env_vars, expected in env_vars_expected:
+                    with unittest.mock.patch("os.environ", env_vars):
+                        self.assertEqual(_colorize.can_colorize(), expected)
+
+                isatty_mock.return_value = False
```

Comment:
Sorry for not being clear :(

I think basically _just_ this test should go in `test__colorize.py`, as this is the only one that directly tests `_colorize.py`. ISTM that the other tests are asserting the _use_ of the `_colorize` module by the `traceback` module -- so probably those tests should stay in `test_traceback.py`

Reply:
No problem, updated!